src/tiled/mapeditor.cpp

```diff
@@ -508,6 +492,67 @@ void MapEditor::showMessage(const QString &text, int timeout)
+void MapEditor::updateSpecificTools(AbstractTool *tool)
+{
+    mToolSpecificToolBar->clear();
+
+    if (!tool)
+        return;
+
+    if (tool == mStampBrush || tool == mBucketFillTool) {
+        QIcon diceIcon(QLatin1String(":images/24x24/dice.png"));
+        diceIcon.addFile(QLatin1String(":images/32x32/dice.png"));
+
+        QIcon flipHorizontalIcon(QLatin1String(":images/24x24/flip-horizontal.png"));
+        flipHorizontalIcon.addFile(QLatin1String(":images/32x32/flip-horizontal.png"));
+
+        QIcon flipVerticalIcon(QLatin1String(":images/24x24/flip-vertical.png"));
+        flipVerticalIcon.addFile(QLatin1String(":images/32x32/flip-vertical.png"));
+
+        QIcon rotateLeftIcon(QLatin1String(":images/24x24/rotate-left.png"));
+        rotateLeftIcon.addFile(QLatin1String(":images/32x32/rotate-left.png"));
+
+        QIcon rotateRightIcon(QLatin1String(":images/24x24/rotate-right.png"));
+        rotateRightIcon.addFile(QLatin1String(":images/32x32/rotate-right.png"));
+
+        QToolButton *mRandomButton = new QToolButton(mToolSpecificToolBar);
+        mRandomButton->setIcon(diceIcon);
+        mRandomButton->setCheckable(true);
+        mRandomButton->setToolTip(tr("Random Mode"));
+        mRandomButton->setShortcut(QKeySequence(tr("D")));
+        mRandomButton->setChecked(mStampBrush->random());
+
+        QToolButton *mFlipHorizontalButton = new QToolButton(mToolSpecificToolBar);
+        mFlipHorizontalButton->setIcon(flipHorizontalIcon);
+        mFlipHorizontalButton->setShortcut(QKeySequence(tr("X")));
+
+        QToolButton *mFlipVerticalButton = new QToolButton(mToolSpecificToolBar);
+        mFlipVerticalButton->setIcon(flipVerticalIcon);
+        mFlipVerticalButton->setShortcut(QKeySequence(tr("Y")));
+
+        QToolButton *mRotateLeft = new QToolButton(mToolSpecificToolBar);
+        mRotateLeft->setIcon(rotateLeftIcon);
+        mRotateLeft->setShortcut(QKeySequence(tr("Shift+Z")));
+
+        QToolButton *mRotateRight = new QToolButton(mToolSpecificToolBar);
+        mRotateRight->setIcon(rotateRightIcon);
+        mRotateRight->setShortcut(QKeySequence(tr("Z")));
+
+        mToolSpecificToolBar->addWidget(mRandomButton);
+        mToolSpecificToolBar->addWidget(mFlipHorizontalButton);
+        mToolSpecificToolBar->addWidget(mFlipVerticalButton);
+        mToolSpecificToolBar->addWidget(mRotateLeft);
+        mToolSpecificToolBar->addWidget(mRotateRight);
+
+        connect(mRandomButton, &QToolButton::toggled, mStampBrush, &StampBrush::setRandom);
+        connect(mRandomButton, &QToolButton::toggled, mBucketFillTool, &BucketFillTool::setRandom);
+        connect(mFlipHorizontalButton, &QToolButton::clicked, this, &MapEditor::flipHorizontally);
+        connect(mFlipVerticalButton, &QToolButton::clicked, this, &MapEditor::flipVertically);
+        connect(mRotateLeft, &QToolButton::clicked, this, &MapEditor::rotateRight);
+        connect(mRotateRight, &QToolButton::clicked, this, &MapEditor::rotateLeft);
```

Comment:
Hmm..looks like I will need to store all the objects too in a list probably and use `qDeleteAll`